Book-A-Desk.Infrastructure/DynamoDbEventStore.fs

```diff
@@ -0,0 +1,47 @@
+﻿namespace Book_A_Desk.Infrastructure
+
+open System
+open Amazon.DynamoDBv2
+open FSharp.AWS.DynamoDB
+open FSharp.Control
+open Book_A_Desk.Domain
+open Book_A_Desk.Domain.Events
+
+module rec DynamoDbEventStore =
+    
+    type DynamoDbEventStore =
+        {        
+            GetEvents: Guid -> Result<DomainEvent seq, string> Async
+            AppendEvents: Map<Guid, DomainEvent seq> -> unit Async
+        }
+    
+    let provide (dynamoDbClient : IAmazonDynamoDB) =
+        let table =
+            TableContext
+                .Create<DeskBooked>(
+                    dynamoDbClient,
+                    tableName = "ReservationEvent",
+                    createIfNotExists = false)
+        
+        {
+            GetEvents = getEvent table
+            AppendEvents = appendEvent table
+        }
+        
+    let private getEvent table aggregateId = async {
+        let! results = table.QueryAsync(keyCondition = <@ fun (r : DeskBooked) -> r.AggregateId = aggregateId @>)
+        
+        let domainResults = DomainMapper.toDomain results
+        
+        return Result.Ok(domainResults)        
+    }
+    
+    let private appendEvent table events = async {
+        let infraEvents = Map.map (fun _ events -> DomainMapper.toInfra events) events
+        
+        let events = infraEvents
+                   |> EventBatcher.batchEvents
```

Comment:
Events must be passed in batches of at most 25, which is why we batch them. Since this is complicated code, we test it. Ideally, I'd like to have it be part of the Domain, but I'm not sure it's a good idea to introduce it into the Domain, since this batching is an infrastructure-only concern.

Maybe this is one of the exceptions where testing the infrastructure makes sense.